platform/osgi/platform-osgi-conditions/src/test/java/org/codice/ddf/condition/BundleNameConditionTest.java

```diff
@@ -0,0 +1,140 @@
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2/bundle3"}));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(true));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(true));
+  }
+
+  @Test
+  public void testIsNotSatisfied() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle4");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition =
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2/bundle3"}));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(false));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(false));
+  }
+
+  @Test
+  public void testIsNotSatisfiedNoArgs() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle4");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition =
+        new BundleNameCondition(
+            bundle, new ConditionInfo(BundleNameCondition.class.getName(), new String[0]));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(false));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(false));
+  }
+
+  @Test
+  public void testIsAllSatisfied() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle2");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition2 =
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2"}));
+
+    bundle = mock(Bundle.class);
+    headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle1");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition1 =
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2"}));
+
+    boolean satisfied =
+        bundleNameCondition1.isSatisfied(
+            new Condition[] {bundleNameCondition1, bundleNameCondition2}, null);
+    assertThat(satisfied, is(true));
+  }
+
+  @Test
+  public void testIsAllNotSatisfied() {
```

Comment:
❗️ We should add a test where one `BundleNameCondition.isSatisfied()` returns `true` and the other returns `false`.

Reply:
Wouldn't that just be the same thing as the first 2 tests?

Reply:
I don't think so. Right now we're testing when they all match and when none match, but not when there's a partial match. Unless I'm missing something...